src/main/java/org/eclipse/tractusx/ssi/lib/exception/key/InvalidPrivateKeyFormatException.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * ******************************************************************************
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * *******************************************************************************
+ */
+
+package org.eclipse.tractusx.ssi.lib.exception.key;
+
+import org.eclipse.tractusx.ssi.lib.exception.SSIException;
+
+/** The type Invalide private key format. */
+public class InvalidPrivateKeyFormatException extends SSIException {
+  /**
+   * Instantiates a new Invalid private key format.
+   *
+   * @param correctLength the correct length
+   * @param providedLength the provided length
+   */
+  public InvalidPrivateKeyFormatException(int correctLength, int providedLength) {
```

Comment:
Class should only provide this specific constructor for violation of expected key length. The other ones are not used at all...